packages/cmf/src/sagas/http.js

```diff
@@ -286,6 +287,78 @@ export function getDefaultConfig() {
 	return HTTP.defaultConfig;
 }
 
+function* handleHTTPAction({
+	method,
+	type,
+	url,
+	body,
+	transform,
+	onResponse,
+	onError,
+	cmf,
+	...config
+}) {
+	// TODO find a solution to split config, options, cmf related
+	const options = {};
+	let response;
+	if (type === HTTP_METHODS.DELETE) {
+		response = yield call(httpDelete, url, config, options);
+	} else if (type === HTTP_METHODS.GET) {
+		response = yield call(httpGet, url, config, options);
+	} else if (type === HTTP_METHODS.POST) {
+		response = yield call(httpPost, url, body, config, options);
+	} else if (type === HTTP_METHODS.PUT) {
+		response = yield call(httpPut, url, body, config, options);
+	} else if (type === HTTP_METHODS.PATCH) {
+		response = yield call(httpPatch, url, body, config, options);
+	}
+	if (!response.response.ok) {
+		if (typeof onError === 'function') {
+			yield put(onError(response.response));
+		} else if (typeof onError === 'string') {
+			yield put({
+				type: onError,
+				error: response.response,
+			});
+		} else {
+			yield put({
+				type: ACTION_TYPE_HTTP_ERRORS,
+				error: response.response,
+			});
+		}
+	} else {
+		let data = response.data;
+		if (transform) {
+			data = transform(data);
+		}
+		if (typeof onResponse === 'function') {
+			yield put(onResponse(data));
+		} else if (typeof onResponse === 'string') {
+			yield put({
+				type: onResponse,
+				response: response.data,
+			});
+		}
+		if (cmf) {
+			if (cmf.collectionId) {
+				yield put(actions.collections.addOrReplace(cmf.collectionId, data));
+			}
+		}
+	}
+}
+
+/**
+ * saga forked at bootstrap to take all actions.http signal
+ */
+function* takeAllHTTPAction() {
+	yield takeEvery(HTTP_METHODS.DELETE, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.GET, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.POST, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.PUT, handleHTTPAction);
+	yield takeEvery(HTTP_METHODS.PATCH, handleHTTPAction);
+	yield take('DO_NOT_QUIT');
+}
```

Comment:
Could be replaced by : 

```
function* takeAllHTTPAction() {
	yield takeEvery(
		[
			HTTP_METHODS.DELETE,
			HTTP_METHODS.GET,
			HTTP_METHODS.POST,
			HTTP_METHODS.PUT,
			HTTP_METHODS.PATCH
		],
		handleHTTPAction
	);
	yield take('DO_NOT_QUIT');
}
```

https://github.com/redux-saga/redux-saga/blob/f85674b3a319c1f9c209e62bc509a0e2c0b4bdeb/packages/core/src/internal/proc.js#L448

https://github.com/redux-saga/redux-saga/blob/d98344e4fd6a7be561bd26f58d5c831e3b14c1b8/packages/core/src/internal/matcher.js#L14



... Even if I'm not sure to understand the goal of the last `take` in this case.